app/src/main/java/ru/practicum/android/diploma/global/data/network/dto/RetrofitNetworkClient.kt

```diff
@@ -0,0 +1,39 @@
+package ru.practicum.android.diploma.global.data.network.dto
+
+import android.util.Log
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import retrofit2.HttpException
+import ru.practicum.android.diploma.global.data.network.HhApi
+import ru.practicum.android.diploma.global.data.network.NetworkClient
+import ru.practicum.android.diploma.global.util.NetworkUtil
+import ru.practicum.android.diploma.global.util.RequestResult
+import ru.practicum.android.diploma.global.util.ResponseCodes
+
+class RetrofitNetworkClient(
+    private val hhApi: HhApi,
+    private val networkUtil: NetworkUtil
+
+) : NetworkClient {
+    override suspend fun doRequest(request: Request): RequestResult<Response> {
+        if (networkUtil.isConnected()) return RequestResult.Error(ResponseCodes.CODE_NO_CONNECT)
```

Comment:
```suggestion
        if (networkUtil.isConnected()) { 
             return RequestResult.Error(ResponseCodes.CODE_NO_CONNECT)
        }
```

И лучше не использовать сокращения в названиях ошибок